parser/prism/Parser.h

```diff
@@ -30,7 +30,7 @@ class ParseError {
 
 class Parser final {
     // The version of Ruby syntax that we're parsing with Prism. This determines what syntax is supported or not.
-    static constexpr std::string_view ParsedRubyVersion = "3.3.0";
+    static constexpr std::string_view ParsedRubyVersion = "3.4.0";
```

Comment:
I wasn't sure about this. This Sorbet support running in a legacy mode, for older code bases?

Reply:
This is necessary for the specs in this PR to pass (otherwise `it` is treated as a method call).

For instance, the AST of `it_param_basic_location_test` shows:
```
"params" : { "type" : "NumParams", "name" : "it" }  // 'it' is a parameter
```
If we revert this to "3.3.0", instead we get:
```
"params" : null                                    // No parameters!
"receiver" : { "type" : "Send", "method" : "it" }  // 'it' is a method call
```

(It could possibly be helpful for this to break on Ruby 3.3 code, if this surfaces call sites that need to be changed for Ruby 3.4 compatibility.)

Reply:
Sorbet's legacy mode for older versions of Ruby are the older versions of Sorbet.